### PR TITLE
Simpler serverlist search

### PR DESCRIPTION
**Pull request by red-001**
#4472 but without the ability to filter the results.

### CODE REVIEW

File: src/util/string.h [hunk 1]
```diff
@@ -272,7 +272,7 @@ inline bool is_yes(const std::string &str)
 {
 	std::string s2 = lowercase(trim(str));
 
-	return s2 == "y" || s2 == "yes" || s2 == "true" || atoi(s2.c_str()) != 0;
+	return s2 == "y" || s2 == "yes" || s2 == "true" || s2 == "on" || atoi(s2.c_str()) != 0;
```
Comment (sfan5) [inline src/util/string.h]: what is this needed for again?

Comment (SmallJoker) [inline src/util/string.h]: For the extended part of this PR with filters like `pvp:on` but right now is this change pointless.

Comment (red-001) [inline src/util/string.h]: Should I make a separate PR for it? It's a fairly trivial modification.

File: builtin/mainmenu/tab_multiplayer.lua [hunk 2]
```diff
@@ -199,16 +230,75 @@ local function main_button_handler(tabview, fields, name, tabdata)
...
+		tabdata.search_for = fields.te_search
+
+		if #menudata.favorites < 2 then 
+			return true 
+		end
+
+		menudata.search_result = {}
+
+		-- setup the keyword list
+		local keywords = {}
```
Comment (rubenwardy) [inline builtin/mainmenu/tab_multiplayer.lua]: I'm wondering where it's worth generalising this code so that it can be reused by #4806

File: src/util/string.h (hunk 1 above)
Comment (sfan5) [inline src/util/string.h]: Please remove it from this PR

File: builtin/mainmenu/tab_multiplayer.lua (hunk 2 above)
Comment (red-001) [inline builtin/mainmenu/tab_multiplayer.lua]: That would probably only make it the PR more complicated. I doubt there is a lot of code that can be used by both PR's.

**Comment by rubenwardy** [comment]
:+1:

**Comment by sofar** [comment]
I... don't see a search box appear when I apply this patch?

I assume it's supposed to show in the `client` tab, right?

**Comment by sofar** [comment]
:+1:

I've fixed some whitespace in my local branch. I'll push if no further objections.

**Comment by sofar** [comment]
f3bd4c405
